crates/cli-tools/src/changelog.rs

```diff
@@ -68,32 +68,37 @@ impl Changelog {
         fs::write(self.changelog_path(), self.to_string().as_bytes()).await
     }
 
-    fn push_description(&mut self, severity: Severity, content: &str) -> Result<()> {
-        let content = if content.starts_with("- ") { content } else { &format!("- {content}") };
-        let current_release = self.get_or_create_release_mut();
-        current_release.push_content(severity, content)
+    fn get_current_release(&self) -> &Release {
+        self.releases.first().unwrap()
+    }
+
+    fn get_current_release_mut(&mut self) -> &mut Release {
+        self.releases.first_mut().unwrap()
     }
 
-    // Gets newest (first) release. Creates a new one if newest release is not prerelease.
-    fn get_or_create_release_mut(&mut self) -> &mut Release {
-        let current_release = self.releases.first().unwrap();
+    fn push_description(&mut self, severity: Severity, content: &str) -> Result<()> {
+        let content = if content.starts_with("- ") { content } else { &format!("- {content}") };
 
-        // Current version is released, insert a new one.
-        if current_release.version.pre.is_empty() {
-            let mut next_version = Version::new(
-                current_release.version.major,
-                current_release.version.minor,
-                current_release.version.patch + 1,
-            );
+        let current_release = self.get_current_release();
+        let is_new_release_needed =
+            current_release.version.pre.is_empty() || current_release.get_max_severity() > severity;
```

Comment:
This doesn't sound right to me, I would expect only the `pre.is_empty()` check to be needed. In the other case, we don't want a new release but just to call `increment_version()`.